agent.go

```diff
@@ -1031,6 +1031,12 @@ func (p *pod) runContainerProcess(cid, pid string, terminal bool, started chan e
 		return err
 	}
 
+	// Make sure all output has been routed before to go further and
+	// return the exit code.
+	ctr.closeProcessStreams(pid)
```

Comment:
@sboeuf do we need the closeProcessStreams? Its been called in defer.

Reply:
If we don't, `wgRouteOutput.Wait()` would be stuck forever. The design is not the prettiest for this function but it works.
And the defer is not a problem cause if the streams have been closed and set to nil, they won't be closed again.
We could (actually should) improve this code.